tests/unit-tests/admin/test-class-sensei-course-pre-publish-panel.php

```diff
@@ -98,9 +97,146 @@ public function testMaybePublishLessons_SufficientPermissionsAndMetaIsTrue_DoesP
+		Sensei_Course_Pre_Publish_Panel::instance()->maybe_publish_lessons( $this->course_id, null, 'draft' );
+
+		// This mimics a new unsaved lesson that just became a saved one in draft state via the Course structure save call. This needs to get published in the next call.
+		$unsaved_to_draft_lesson_id = $this->factory->lesson->create(
+			[
+				'post_status' => 'draft',
+				'meta_input'  => [
+					'_lesson_course' => $this->course_id,
+				],
+			]
+		);
+
+		/* Act */
+		// Notice how the 'old_status' is 'publish' here. Because after publishing the post and the structure is saved, the old status is 'publish' for that Course,
+		// because the Course already got published in the previous call.
+		// Our Course publishing sequence from Gutenberg is like this:
+		// GB sends Publish Course call -> Then we send the structure saving call -> Then we send a Course update call.
+		Sensei_Course_Pre_Publish_Panel::instance()->maybe_publish_lessons( $this->course_id, null, 'publish' );
+
+		/* Assert */
+		$this->assertEquals( 'publish', get_post_status( $unsaved_to_draft_lesson_id ) );
+	}
+
+	/**
+	 * In the first subsequent call, the lessons are published, but not in the second or more subsequent calls.
+	 *
+	 *  @covers Sensei_Course_Pre_Publish_Panel::maybe_publish_lessons
+	 */
+	public function testMaybePublishLessons_AfterFirstPublishSequence_FartherSubsequentCallsDoNotPublishLessons() {
+		/* Arrange */
+		// Check the comments in the previous test for the explanation of the testing.
+		$this->login_as_admin();
+		update_post_meta( $this->course_id, 'sensei_course_publish_lessons', true );
+
+		Sensei_Course_Pre_Publish_Panel::instance()->maybe_publish_lessons( $this->course_id, null, 'draft' );
+
+		$unsaved_to_draft_lesson_id = $this->factory->lesson->create(
+			[
+				'post_status' => 'draft',
+				'meta_input'  => [
+					'_lesson_course' => $this->course_id,
+				],
+			]
+		);
+		Sensei_Course_Pre_Publish_Panel::instance()->maybe_publish_lessons( $this->course_id, null, 'publish' );
+
+		$unsaved_to_draft_lesson_id_2 = $this->factory->lesson->create(
+			[
+				'post_status' => 'draft',
+				'meta_input'  => [
+					'_lesson_course' => $this->course_id,
+				],
+			]
+		);
+		Sensei_Course_Pre_Publish_Panel::instance()->maybe_publish_lessons( $this->course_id, null, 'publish' );
+
+		$unsaved_to_draft_lesson_id_3 = $this->factory->lesson->create(
+			[
+				'post_status' => 'draft',
+				'meta_input'  => [
+					'_lesson_course' => $this->course_id,
+				],
+			]
+		);
+
+		/* Act */
+		// See comments in the previous test for the explanation of the 'old_status'.
+		Sensei_Course_Pre_Publish_Panel::instance()->maybe_publish_lessons( $this->course_id, null, 'publish' );
 
 		/* Assert */
 		$this->assertEquals( 'publish', get_post_status( $this->lesson_id ) );
+		$this->assertEquals( 'publish', get_post_status( $unsaved_to_draft_lesson_id ) );
```

Comment:
When we have multiple assertions, it's best to add a short description as the third param so that we know which test failed.

Reply:
Good idea, added here https://github.com/Automattic/sensei/pull/7582/commits/137a1050661f039056fd5ef8f4c7a8937e59e98f